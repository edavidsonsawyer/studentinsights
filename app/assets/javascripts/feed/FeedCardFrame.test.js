import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FeedCardFrame from './FeedCardFrame';
import {withDefaultNowContext} from '../testing/NowContainer';
import PerDistrictContainer from '../components/PerDistrictContainer';


function testStudent() {
  return {
    id: 55,
    first_name: 'Mari',
    last_name: 'Skywalker',
    grade: '9',
    house: 'Beacon',
    school: {
      local_id: 'SHS',
      school_type: 'HS'
    },
    homeroom: {
      id: 13,
      name: 'SHS-052',
      educator: {
        id: 5,
        email: 'lt@demo.studentinsights.org',
        full_name: 'Teacher, Lois',
      }
    }
  };
}

function testStudentElementary() {
  return {
    id: 32,
    first_name: 'Pluto',
    last_name: 'Skywalker',
    grade: '3',
    house: null,
    school: {
      local_id: 'HEA',
      school_type: 'ESMS'
    },
    homeroom: {
      id: 32,
      name: 'HEA-011',
      educator: {
        id: 3,
        email: 'alex@demo.studentinsights.org',
        full_name: 'Teacher, Alex',
      }
    }
  };
}

function testProps(student) {
  return {
    byEl: <div>by</div>,
    whereEl: <div>where</div>,
    whenEl: <div>when</div>,
    badgesEl: <div>badges</div>,
    children: 'kids',
    student
  };
}

function testEl(props = {}, context = {}) {
  const districtKey = context.districtKey || 'somerville';
  return withDefaultNowContext(
    <PerDistrictContainer districtKey={districtKey}>
      <FeedCardFrame {...props} />
    </PerDistrictContainer>
  );
}

function testRender(props = {}, context = {}) {
  const el = document.createElement('div');
  ReactDOM.render(testEl(props, context), el);
  return el;
}

it('renders for HS without crashing', () => {
  const el = testRender(testProps(testStudent()));
  expect($(el).text()).toContain('Mari Skywalker');
  expect($(el).text()).toContain('9th grade');
  expect($(el).text()).not.toContain('SHS-052');
  expect($(el).text()).not.toContain('with Lois Teacher');
  expect($(el).text()).toContain('by');
  expect($(el).text()).toContain('where');
  expect($(el).text()).toContain('when');
  expect($(el).text()).toContain('badges');
  expect($(el).text()).toContain('kids');
});

it('renders homeroom for ESMS', () => {
  const el = testRender(testProps(testStudentElementary()));
  expect($(el).text()).toContain('HEA-011');
  expect($(el).text()).toContain('with Alex Teacher');
});


it('matches all snapshots', () => {
  const tree = renderer
    .create(
      <div>
        {testEl(testProps(testStudent()))}
        {testEl(testProps(testStudentElementary()))}
      </div>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
