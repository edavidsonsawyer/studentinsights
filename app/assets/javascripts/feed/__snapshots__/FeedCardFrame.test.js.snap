// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches all snapshots 1`] = `
<div>
  <div
    className="Card FeedCardFrame"
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div
      className="FeedCardFrame-FrameHeader"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
        >
          <div>
            <a
              href="/students/55"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Mari
               
              Skywalker
            </a>
          </div>
          <div>
            9th grade
          </div>
          <div />
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
            "textAlign": "right",
          }
        }
      >
        <div>
          <div>
            by
          </div>
        </div>
        <div>
          <div>
            where
          </div>
        </div>
        <div>
          <div>
            when
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      kids
    </div>
    <div
      className="FeedCardFrame-footer"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <div />
      <div>
        badges
      </div>
    </div>
  </div>
  <div
    className="Card FeedCardFrame"
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div
      className="FeedCardFrame-FrameHeader"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
        >
          <div>
            <a
              href="/students/32"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pluto
               
              Skywalker
            </a>
          </div>
          <div>
            3rd grade
          </div>
          <div>
            <span
              className="Homeroom"
              style={Object {}}
            >
              <a
                href="/homerooms/32"
              >
                HEA-011
              </a>
              <span>
                <span>
                   
                </span>
                 with 
                <a
                  className="Educator"
                  href="mailto:alex@demo.studentinsights.org"
                  style={Object {}}
                >
                  Alex Teacher
                </a>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
            "textAlign": "right",
          }
        }
      >
        <div>
          <div>
            by
          </div>
        </div>
        <div>
          <div>
            where
          </div>
        </div>
        <div>
          <div>
            when
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      kids
    </div>
    <div
      className="FeedCardFrame-footer"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <div />
      <div>
        badges
      </div>
    </div>
  </div>
</div>
`;
